src/libcore/ptr.rs

```diff
@@ -2509,6 +2509,29 @@ pub fn eq<T: ?Sized>(a: *const T, b: *const T) -> bool {
     a == b
 }
 
+/// Hash the raw pointer address behind a reference, rather than the value
+/// it points to.
+///
+/// # Examples
+///
+/// ```
+/// use std::collections::hash_map::DefaultHasher;
+/// use std::hash::Hasher;
+/// use std::ptr;
+///
+/// let five = 5;
+/// let five_ref = &five;
+///
+/// let mut hasher = DefaultHasher::new();
+/// ptr::hash(five_ref, &mut hasher);
+/// println!("Hash is {:x}!", hasher.finish());
```

Comment:
Prefer using `assert_eq!` here so that this acts as a test against the implementation.

Reply:
Gladly, but is there a way to get a stable pointer hash to assert against?  I found the println in another doc example for hash so I assumed there wasn't.

Reply:
I think you should just be able to use `(five_ref as *const T).hash(&mut other_hasher)`, then `finish` and assert_eq with that as the expected value? This is what `NonNull::from(a).hash(&mut into)` does anyways... it converts to a raw pointer internally and then the hash method just delegates to that of the pointer.

Reply:
Sure, I can do that.